test/unit/get-branch-test.js

```diff
@@ -41,3 +41,26 @@ test('get branch request succeeds', t => {
 
   .catch(t.error)
 })
+
+test('get branch test fails', t => {
+  const state = {
+    api,
+    debug: () => {}
+  }
+
+  simple.mock(api.repos, 'getBranch').rejectWith({
+    code: 404
+  })
+
+  getBranch(state)
+
+  .then(() => {
+    t.error('should not resolve')
```

Comment:
oh I think this should not be `t.error` but `t.fail`, but I mix the two up all the time myself. It will probably work like that, too. `t.error` should get an `error` instance, it's meant to be used in promises like this

```
doSomething()

.then(() => {
  // ...
})

.catch(t.error)
```

or callbacks like this

```
doSomething(error => {
  t.error(error)
  // ...
})

If `error` is undefined, `t.error` does nothing, otherwise it makes the test fail and shows the error it received.

So `t.error` is an assertion, while `t.fail` is a method that directly makes the test fail, without any assertion.